client/src/js/services/dropdownBehavior.js

```diff
@@ -0,0 +1,29 @@
+angular.module('bhima.services')
+.service('DropdownBehaviorService', DropdownBehaviorService);
+
+function DropdownBehaviorService() {
+
+  var service = this;
+
+  //this function check if html element's position is on the bottom
+  // if true it changes the element's position in order to display it conviniently
+
+  // it function is usefull for carets for exemple
+  // the Users page dropdown won't falls off page thanks to this function
+  // it takes the dropdown id as parameter
+
+  service.setPosition = function(id_dropdown){
+    
+    var menu =  $('#'+ id_dropdown);
```

Comment:
In general, you should try not use JQuery when you use AngularJS.  They have two different ways of tackling problems.

Instead of JQuery, AngularJS provides [jqLite (also known as angular.element())](https://docs.angularjs.org/api/ng/function/angular.element) to manipulate the DOM.  In general, AngularJS also asks that you do DOM manipulations in a directive.

I propose that you move this into a directive that extends Bootstrap's dropdown menu.  It would listen on the [isOpen](https://github.com/angular-ui/bootstrap/blob/master/src/dropdown/dropdown.js#L239) variable, and then position the dropdown menu appropriately.  You can still use a service to do the calculation, but all DOM manipulation should happen in a directive.

It would something like this:
```html
<ul uib-dropdown-menu  bh-dropdown-menu-auto-dropup>
</ul>
```